core/controllers/reader.py

```diff
@@ -244,6 +244,20 @@ def get(self, exploration_id):
         self.render_json(self.values)
 
 
+class StatsEventsHandler(base.BaseHandler):
+    """Handles a batch of events coming in from the frontend."""
+
+    REQUIRE_PAYLOAD_CSRF_CHECK = False
+
+    @acl_decorators.can_play_exploration
+    def post(self, exploration_id):
+        event_dicts = self.payload.get('event_dicts')
```

Comment:
Validate for the event dicts? (Raise 400 if invalid.)

Reply:
Done.